salt/modules/logmod.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+'''
+On-demand logging
+=================
+
+.. versionadded:: Nitrogen
+
+The sole purpose of this module is logging messages in the (proxy) minion.
+It comes very handy when debugging complex Jinja templates, for example:
+
+.. code-block:: jinja
+
+    {%- for var in range(10) %}
+      {%- set logged = salt.log.info(var) %}
+    {%- endfor %}
+
+CLI Example:
+
+.. code-block:: bash
+
+    salt '*' log.error 'Please dont do that, this module is not for CLI use!'
```

Comment:
Yes, sorry for being ambiguous.

I am proposing this module with the sole purpose to be called from inside the Jinja templates (and potentially other engines), e.g.: `{% do salt.log.error('test message') %`.
At the same time, the user is able to do on the CLI:
```bash
salt '*' log.error 'message from the CLI'
```
Which will log artificial log messages as well.
While logging from Jinja has a point, logging from the CLI it does not and I was thinking if we could force to not log when this is called from the CLI.

I have thought about the following:
```python
def debug(message, **kwargs):
    '''Log message at level DEBUG.'''
    if not kwargs:
        # log only when not executed from the CLI
        log.debug(message)
        return True
    return False
```
Which does not log when executed from the CLI because in the `kwargs` are set the `__pub_user`, `__pub_jid` keys.

Is this anything better?